url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/connection/config/DefaultConnectionConfiguration.java

```diff
@@ -0,0 +1,152 @@
+package ping.pong.net.connection.config;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author mfullen
+ */
+public class DefaultConnectionConfiguration implements ConnectionConfiguration
+{
+    /**
+     *
+     */
+    public static final Logger LOGGER = LoggerFactory.getLogger(DefaultConnectionConfiguration.class);
+
+    /**
+     *
+     * @param filepath
+     * @return
+     */
+    public static String findKeyStorePath(String filepath)
+    {
+        if (filepath == null)
+        {
+            return getKeystoreFromClassPath();
+        }
+
+        File file = new File(filepath);
+        if (file.exists())
+        {
+            return file.getAbsolutePath();
+        }
+
+        return getKeystoreFromClassPath();
+    }
+
+    private static String getKeystoreFromClassPath()
+    {
+        String path = null;
+        try
+        {
+            path = new File(Thread.currentThread().getContextClassLoader().getResource(DEFAULT_KEY_STORE).toURI()).getAbsolutePath();
+            LOGGER.trace("Couldn't find specified keystore, reverting to default {}", path);
+        }
+        catch (URISyntaxException ex)
+        {
+            LOGGER.error("URI error.", ex);
+        }
+        return path;
+    }
+
+    DefaultConnectionConfiguration()
+    {
+        this(DEFAULT_TCP_PORT, DEFAULT_UDP_PORT, LOCAL_HOST, false);
+    }
+
+    DefaultConnectionConfiguration(int port, int udpPort, String ipAddress, boolean ssl)
+    {
+        this(port, udpPort, ipAddress, ssl, ssl ? getKeystoreFromClassPath() : DEFAULT_KEY_STORE, DEFAULT_KEY_STORE_PASSWORD);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. Hardcoded secrets or credentials can allow attackers to bypass authentication methods and perform malicious actions. We recommend revoking access to resources using this credential and storing future credentials in a management service such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

[Learn more about the use of hardcoded credentials](https://cwe.mitre.org/data/definitions/798.html)